### DISCUSSION TITLE
Server NetworkObject pooling scene stacking.

### DISCUSSION THREAD

**Discussion by duyca** [FirstGearGames/FishNet#483] [Feature Request]
My game is 1vs1 battle. Then room Scene Load and Unload happen with high frequency. But NetworkObject (NO) in that scene got destroy and re-create whenever Scene Load and Unload. It make server GC heavy effect on performance.
Should we extend pooling for scene stacking as well.

**Comment by FirstGearGames** [comment]
It would be difficult to automate the process because some devs might prefer instantiated objects to be destroyed in scenes, even if they are set to pool with despawn. My recommendation is to hook into one of the SceneManager events that fire before unloading and pool them there.
